### PR TITLE
feat(web): navigate with keyboard on person page

### PR DESCRIPTION
**Pull request by martabal**
## Changes made in this PR

With this PR, users can navigate through suggested people with Tab and Arrow keys. On the suggestion merge modal, pressing the Tab key will only switch between the `No` and `Yes` buttons.

Resolves #5400 and #5398

### CODE REVIEW

**Review by jrasm91 — COMMENTED** [review]
Can you use svelte:document instead please?

https://svelte.dev/docs/special-elements#svelte-document

File: web/src/lib/components/elements/buttons/button.svelte [hunk 1]
```diff
@@ -26,6 +26,7 @@
   export let fullwidth = false;
   export let border = false;
   export let title: string | undefined = '';
+  export let ref: HTMLButtonElement | null = null;
```
Comment (jrasm91) [inline web/src/lib/components/elements/buttons/button.svelte:29]: What is this used for?

File: web/src/lib/components/faces-page/merge-suggestion-modal.svelte [hunk 2]
```diff
@@ -8,27 +8,57 @@
...
 
+  const dispatch = createEventDispatcher<{
+    reject: void;
+    confirm: [PersonResponseDto, PersonResponseDto];
+    close: void;
+  }>();
+
+  const handleKeyboardPress = (event: KeyboardEvent) => {
+    if (!$showAssetViewer) {
+      event.stopPropagation();
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/merge-suggestion-modal.svelte:33]: Why are you stopping propagation, especially if you aren't handling the keyboard event?

File: web/src/lib/components/faces-page/merge-suggestion-modal.svelte [hunk 3]
```diff
@@ -8,27 +8,57 @@
...
+      switch (event.key) {
+        case 'Tab':
```
Comment (jrasm91) [inline web/src/lib/components/faces-page/merge-suggestion-modal.svelte:35]: Tabbing through focusable elements is something that is built into the browser. We don't need to re-invent the wheel here.

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

Comment (martabal) [reply web/src/lib/components/faces-page/merge-suggestion-modal.svelte:35]: Yeah I know but I couldn't find a way to limit the focus to the suggested people only.

File: web/src/lib/components/elements/buttons/button.svelte (hunk 1 above)
Comment (martabal) [reply web/src/lib/components/elements/buttons/button.svelte]: I used it to focus on the button from the parent component.

File: web/src/lib/components/faces-page/merge-suggestion-modal.svelte (hunk 3 above)
Comment (jrasm91) [reply web/src/lib/components/faces-page/merge-suggestion-modal.svelte:34]: tabindex=-1 tells the browser to skip it.

Comment (martabal) [reply web/src/lib/components/faces-page/merge-suggestion-modal.svelte:34]: Yeah I tried it but the issue is that you want to navigate only through the suggested people, not all the buttons on the page. Do you have any ideas :thinking: ?

Comment (jrasm91) [reply web/src/lib/components/faces-page/merge-suggestion-modal.svelte:34]: Does using an actual browser model still let you focus on background elements?

Comment (martabal) [reply web/src/lib/components/faces-page/merge-suggestion-modal.svelte:34]: What's an actual browser model ?

Comment (martabal) [reply web/src/lib/components/faces-page/merge-suggestion-modal.svelte:34]: Mmmh, just tried it, does it work only for dialogs ?

Comment (martabal) [reply web/src/lib/components/faces-page/merge-suggestion-modal.svelte:34]: I don't think we want dialogs to display suggested people :sweat_smile:

**Comment by jrasm91** [comment]
I think it is fine to add keyboard shortcuts, but we should do it in a way that doesn't make the site less accessible (which is the outcome of overriding the tab behavior or preventing other default key behaviors)

**Comment by martabal** [comment]
> I think it is fine to add keyboard shortcuts, but we should do it in a way that doesn't make the site less accessible (which is the outcome of overriding the tab behavior or preventing other default key behaviors)

I agree 100% with you. But in this case we want to navigate with the tabs and arrow keys which is a little unusual. I don't know what's the good way for that case :sweat_smile:

**Review by jrasm91 — APPROVED** [review]
LGTM

**Pull request merged**